src/epochs.lua

```diff
@@ -598,25 +586,26 @@ end
 function epochs.distributeRewardsForEpoch(currentTimestamp)
 	local epochIndex = epochs.getEpochIndexForTimestamp(currentTimestamp - epochs.getSettings().durationMs) -- go back to previous epoch
 	local epoch = epochs.getEpoch(epochIndex)
-	if not next(epoch) then
+	if not epoch then
 		-- silently return
-		print("Not distributing rewards for last epoch. New epoch has not been created yet.")
+		print("Unable to distribute rewards for epoch. Epoch not found: " .. epochIndex)
 		return
 	end
 
 	--- The epoch was already distributed
 	--- @cast epoch DistributedEpoch
 	if epoch.distributions.distributedTimestamp then
-		print("Rewards already distributed for epoch: " .. epochIndex)
-		return -- silently return
+		print("Rewards already distributed for epoch. Epoch will be removed from the epoch registry: " .. epochIndex)
+		Epochs[epochIndex] = nil
+		return epoch
```

Comment:
this mean the cleanup didn't happen, so do it here and remove the distributed epoch.

Reply:
Does returning the epoch rather than nil have any unintended downstream repercussions?

Reply:
it means a send-distribution-notice is sent, which is what we want if the epoch was not previously cleared